apps/emqx_auth_http/src/emqx_auth_http.erl

```diff
@@ -83,7 +84,9 @@ authenticate(#http_request{url = Url,
              ClientInfo, HttpHeaders, HttpOpts, RetryOpts) ->
    request(Method, ContentType, Url, feedvar(Params, ClientInfo), HttpHeaders, HttpOpts, Options, RetryOpts).
 
--spec(is_superuser(undefined | #http_request{method::'get' | 'post',params::[any()]}, #{'clientid':=_, 'peerhost':=_, 'protocol':=_, _=>_}, Headers::any(), HttpOpts::any(), #{'backoff':=_, 'interval':=_, 'times':=_, _=>_}) -> boolean()).
+-type http_request() :: #http_request{method::'get' | 'post',params::[any()]}.
+
+-spec(is_superuser(undefined | http_request(), #{'clientid':=_, 'peerhost':=_, 'protocol':=_, _=>_}, Headers::any(), HttpOpts::any(), #{'backoff':=_, 'interval':=_, 'times':=_, _=>_}) -> boolean()).
```

Comment:
since you are here. maybe add a spec for the second arg too.

```
-spec http_opts() :: #{clientid:=_, peerhost:=_, protocol:=_, _=>_}.
-spec retry_opts() :: #{backoff:=_, interval:=_, times:=_, _=>_}.
```